pkg/virt-controller/services/template.go

```diff
@@ -97,6 +99,30 @@ func GetImagePullPolicy(store cache.Store) (policy k8sv1.PullPolicy, err error)
 	return
 }
 
+// return true if a PersistentVolumeClaim uses VolumeMode = Block
+func isPVCBlock(namespace string, claimName string) (bool, error) {
+	// FIXME: using a client to make synchronous calls is suboptimal
+	// ideally this would be using a cache/informer
+	client, err := kubecli.GetKubevirtClient()
+	if err != nil {
+		return false, err
+	}
+	opts := metav1.GetOptions{}
+	pvc, err := client.CoreV1().PersistentVolumeClaims(namespace).Get(claimName, opts)
```

Comment:
Can you introduce an informer which watches for all PersistentVolumeClaims in the cluster? Everyone of us is just writing that we should do it properly. Someone has to start doing it properly.

Reply:
ok

Reply:
I'll take the blame for the synchronous call here. I had intended to switch to an informer once something end-to-end was working, but never got to that point.

Reply:
PS: done